mlflow/pyfunc/__init__.py

```diff
@@ -2429,12 +2429,16 @@ def predict(model_input: List[str]) -> List[str]:
                 CHAT_MODEL_INPUT_SCHEMA,
                 CHAT_MODEL_OUTPUT_SCHEMA,
             )
-            input_example = CHAT_MODEL_INPUT_EXAMPLE
+            input_example = input_example or CHAT_MODEL_INPUT_EXAMPLE
 
-            # perform output validation and throw if
-            # output is not coercable to ChatResponse
-            messages = [ChatMessage(**m) for m in input_example["messages"]]
-            params = ChatParams(**{k: v for k, v in input_example.items() if k != "messages"})
+            if isinstance(input_example, list) and isinstance(input_example[0], ChatMessage):
```

Comment:
Why do we just check first element is ChatMessage not all(isinstance(x, ChatMessage) for x in input_example)?

Reply:
I see, this is an example input. The number of items is probably slow.